oclc/classify.py

```diff
@@ -769,6 +951,75 @@ def lookup_by(self, **kwargs):
         representation, cached = Representation.get(self._db, url)
         return representation.content
 
+class OCLCLookupCoverageProvider(MetadataWranglerBibliographicCoverageProvider):
+
+    def __init__(self, collection, api=None, **kwargs):
+        super(OCLCLookupCoverageProvider, self).__init__(
+            collection, registered_only=False, **kwargs
+        )
+        self.api = api or OCLCClassifyAPI(self._db)
+
+
+class IdentifierLookupCoverageProvider(OCLCLookupCoverageProvider):
+    """Does identifier (specifically, ISBN) lookups using OCLC Classify.
+    """
+    SERVICE_NAME = "OCLC Classify Identifier Lookup"
+    INPUT_IDENTIFIER_TYPES = [Identifier.ISBN]
+    DATA_SOURCE_NAME = DataSource.OCLC
+
+    parser = OCLCClassifyXMLParser()
+
+    def process_item(self, identifier):
+        """Ask OCLC Classify about a single ISBN and create an Edition
+        based on what it says.
+        """
+        # Perform a title/author lookup.
+
+        metadata_list = []
+
+        try:
+            xml = self.api.lookup_by(isbn=identifier.identifier)
+            tree, code, owi_numbers = self.parser.initial_look_up(self._db, xml)
+
+            if code in [0, 2]:
+                metadata_list = self._single(self._db, tree, identifier)
+            elif code == 4:
+                metadata_list = self._multiple(self._db, owi_numbers, identifier)
+
+            for metadata in metadata_list:
+                self._apply(metadata, identifier)
+
+            return identifier
+
+        except IOError as e:
+            return self.failure(identifier, e.message)
+
+    def _single(self, _db, tree, identifier):
+        return [self.parser.parse(self._db, tree, [identifier])]
+
+    def _multiple(self, _db, owi_numbers, identifier):
+        results = []
+        for number in owi_numbers:
+            owi, is_new = Identifier.for_foreign_id(_db, Identifier.OCLC_WORK, number)
+            owi.weight = 1
```

Comment:
This needs to be an IdentifierData, not an Identifier. The database Identifier will be created during the `metadata.apply` call.